pkg/client/block/client.go

```diff
@@ -2,79 +2,28 @@ package block
 
 import (
 	"context"
-	"fmt"
-	"time"
 
 	"cosmossdk.io/depinject"
 
 	"github.com/pokt-network/poktroll/pkg/client"
-	"github.com/pokt-network/poktroll/pkg/either"
-	"github.com/pokt-network/poktroll/pkg/observable"
-	"github.com/pokt-network/poktroll/pkg/observable/channel"
-	"github.com/pokt-network/poktroll/pkg/retry"
+	"github.com/pokt-network/poktroll/pkg/client/events"
 )
 
 const (
-	// eventsBytesRetryDelay is the delay between retry attempts when the events
-	// bytes observable returns an error.
-	eventsBytesRetryDelay = time.Second
-	// eventsBytesRetryLimit is the maximum number of times to attempt to
-	// re-establish the events query bytes subscription when the events bytes
-	// observable returns an error.
-	eventsBytesRetryLimit        = 10
-	eventsBytesRetryResetTimeout = 10 * time.Second
-	// NB: cometbft event subscription query for newly committed blocks.
-	// (see: https://docs.cosmos.network/v0.47/core/events#subscribing-to-events)
+	// committedBlocksQuery is the query used to subscribe to new committed block
+	// events used by the EventsQueryClient to subscribe to new block events from
+	// the chain.
+	// See: https://docs.cosmos.network/v0.47/learn/advanced/events#default-events
 	committedBlocksQuery = "tm.event='NewBlock'"
-	// latestBlockObsvblsReplayBufferSize is the replay buffer size of the
-	// latestBlockObsvbls replay observable which is used to cache the latest block observable.
-	// It is updated with a new "active" observable when a new
-	// events query subscription is created, for example, after a non-persistent
-	// connection error.
-	latestBlockObsvblsReplayBufferSize = 1
-	// latestBlockReplayBufferSize is the replay buffer size of the latest block
-	// replay observable which is notified when block commit events are received
-	// by the events query client subscription created in goPublishBlocks.
-	latestBlockReplayBufferSize = 1
+	replayObsBufferSize  = 1 // the amount of events we want before they are emitted
```

Comment:
Now that the block client exposes `#LastNBlocks()`  instead of `#LatestBlock()`, I think it's prudent that we increase this. I think we should also rename it to `defaultReplayObsBufferSize` and add a TODO: 

```suggestion
	// TODO_TECHDEBT/TODO_FUTURE: add a `blocksLimit` field to the block client struct that defaults to this but can be overridden via an option in future work.
	defaultBlocksLimit  = 1 // the number of blocks that the replay observable returned by LastNBlocks() will be able to replay.
```